### PR TITLE
rofimoji: 6.1.0 -> 6.2.0

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

This PR bumps the version of the `rofimoji` package.
Release logs: https://github.com/fdw/rofimoji/releases/tag/6.2.0

Summary of changes:
- move package to `pkgs/by-name`
- ~remove `with lib;`~
- change `sha256` to `hash`
- use `pyproject = true` instead of `format = "pyproject"` 
- add `meta.changelog`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by justinlovinger** [comment]
Why was `with lib;` removed? This seems a superfluous change for a standard version-bump.

This is my first introduction to RFC 140, so I will defer to a more senior maintainer for whether the change from `all-packages.nix` to `by-name` is appropriate.

I vaguely recall `rev = "refs/tags/${version}"` being a workaround for a bug. If the bug is fixed and `rev = version;` works, that is a good change.

**Comment by TomaSajt** [comment]
> Why was `with lib;` removed? This seems a superfluous change for a standard version-bump.

IMO `with` should only be used when doing stuff like `with lib.maintainers` because everything in the following array will be a part of `lib.maintainers`. Otherwise, I use the longer way to access it, so that it's exactly clear where names are used from.

I can change it back if it's not needed.

> I vaguely recall `rev = "refs/tags/${version}"` being a workaround for a bug. If the bug is fixed and `rev = version;` works, that is a good change.

Related PR: https://github.com/NixOS/nixpkgs/pull/182029
In that PR, the automation was somehow triggered to use the GitHub tag instead of the GitHub release. In this case `refs/tags/version` and `version` point to the same commit, so it doesn't matter which is used.

**Comment by justinlovinger** [comment]
> IMO `with` should only be used when doing stuff like `with lib.maintainers` because everything in the following array will be a part of `lib.maintainers`. Otherwise, I use the longer way to access it, so that it's exactly clear where names are used from.
> 
> I can change it back if it's not needed.

Personally, I agree, but the Nixpkgs convention seems to be to use `with lib;` and the like more freely. I think we should avoid personal style-changes in pull-requests. The Nix formatting RFC may eventually provide a consistent style-guide, <https://github.com/NixOS/rfcs/pull/166>.

**Review by Eisfunke — APPROVED** [review]
Looks good to me. Builds and runs fine on x64.

**Comment by TomaSajt** [comment]
I was reading some discussions about using `python3.pkgs` and it turns out it's not the best for spliced inputs. I am not sure if it would make any difference here, but I moved back to using `inherit (python3Packages)`, but put it into the `package.nix` file instead of `all-packages.nix`.

File: pkgs/by-name/ro/rofimoji/package.nix
Comment (K900) [inline pkgs/by-name/ro/rofimoji/package.nix:17]: This doesn't help, just use `python3Packages` directly if you're doing this.

**Comment by nu-nu-ko** [comment]
> > IMO `with` should only be used when doing stuff like `with lib.maintainers` because everything in the following array will be a part of `lib.maintainers`. Otherwise, I use the longer way to access it, so that it's exactly clear where names are used from.
> > 
> > I can change it back if it's not needed.
> 
> Personally, I agree, but the Nixpkgs convention seems to be to use `with lib;` and the like more freely. I think we should avoid personal style-changes in pull-requests. The Nix formatting RFC may eventually provide a consistent style-guide, <https://github.com/NixOS/rfcs/pull/166>.

https://github.com/NixOS/nixpkgs/issues/208242

there is good reason to avoid this practice and personally dont beleive its ill fit for this PR

**Comment by TomaSajt** [comment]
I have removed the `with lib;` from the optionals, however I left it in for `meta`, because that's a pretty common thing to do.

**Pull request merged**
